### PR TITLE
Trivial changes from string-as-rec

### PR DESCRIPTION
**Pull request by mppf**
This PR is a collection of trivial changes from string-as-rec.  I think
these changes are good, but they are generally tiny code and comment
changes. Each commit describes each individual change.

Passed full local testing.
Reviewed by @lydia-duncan, thanks!

### CODE REVIEW

File: compiler/ifa/num.h [hunk 1]
```diff
@@ -79,6 +79,10 @@ class Immediate { public:
   uint32_t const_kind;
   uint32_t num_index;
   union {
+    // Unions are initalized based off the first element, so we need to have
+    // the largest thing first to make sure it is all zero initalized
+    complex128 v_complex128;
+    complex64  v_complex64;
```
Comment (lydia-duncan) [inline compiler/ifa/num.h]: I get that this is necessary, but it's a little unfortunate that it means booleans and numerical types get intermixed (and that types both increase and decrease in size).  This doesn't need to be changed in this patch, but it's something I wanted to point out while I was thinking about it

Comment (mppf) [inline compiler/ifa/num.h]: Would it be nicer if I moved the float fields below the complex ones?

File: modules/internal/ChapelArray.chpl [hunk 2]
```diff
@@ -704,6 +704,8 @@ module ChapelArray {
     var _value;
     var _valueType;
   
+    // Never, ever create a distribution directly.
+    // Always call _newDistribution() to obtain  one.
```
Comment (lydia-duncan) [inline modules/internal/ChapelArray.chpl]: :+1:

File: compiler/ifa/num.h (hunk 1 above)
Comment (lydia-duncan) [inline compiler/ifa/num.h]: Yeah.  My ideal organization would put ints and uints after that, in decreasing size order.  I think that leaves booleans and strings at the end

**Comment by lydia-duncan** [comment]
These are good changes :+1:

File: compiler/ifa/num.h (hunk 1 above)
Comment (mppf) [inline compiler/ifa/num.h]: OK. I'll re-arrange them in this manner.

**Comment by mppf** [comment]
Running testing...

**Pull request merged**